packages/odie-client/src/components/message/message-content.tsx

```diff
@@ -78,34 +79,13 @@ export const MessageContent = ( {
 							isMessageWithoutEscalationOption={ isMessageWithOnlyText }
 						/>
 					) }
-					{ message.type === 'introduction' && (
-						<div className="odie-introduction-message-content">
-							<div className="odie-chatbox-introduction-message">
-								<Markdown
-									urlTransform={ uriTransformer }
-									components={ {
-										a: CustomALink,
-									} }
-								>
-									{ message.content }
-								</Markdown>
-							</div>
-						</div>
-					) }
-					{ message.type === 'conversation-feedback' && message?.meta?.feedbackUrl && (
-						<div className="odie-introduction-message-content odie-introduction-message-content__conversation_feedback">
-							<p>{ message.content }</p>
-							<p>
-								<a target="_blank" rel="noreferrer" href={ message?.meta?.feedbackUrl }>
-									{ __( 'Submit Rating', __i18n_text_domain__ ) }
-								</a>
-							</p>
-						</div>
-					) }
+					{ message.type === 'introduction' && <IntroductionMessage content={ message.content } /> }
+					{ isFeedbackMessage && <FeedbackMessage content={ message.content } /> }
 					{ ! stopConflatingNegativeRatingWithContactSupport &&
 						message.type === 'dislike-feedback' && <DislikeFeedbackMessage /> }
 				</div>
 			</div>
+			{ isFeedbackMessage && <FeedbackSubmit meta={ message?.meta } /> }
 			{ displayChatWithSupportLabel && <ChatWithSupportLabel /> }
```

Comment:
Thanks Renan, it works great!
Can you please explore if it is possible to have both the content and the submit inside a single component? I would like to reduce the complexity and do not have things spread in multiple places or components, if possible.

Reply:
Thank you for this idea. I have pushed a new commit with a simplified version. 👍 
IMO, it got less complex. :)